bftengine/src/bftengine/ReplicaLoader.cpp

```diff
@@ -337,7 +337,7 @@ ReplicaLoader::ErrorCode loadReplicaData(shared_ptr<PersistentStorage> p, Loaded
       Verify(d.executedSeqNum > ld.lastStableSeqNum, InconsistentErr);
       Verify(d.executedSeqNum <= ld.lastStableSeqNum + kWorkWindowSize, InconsistentErr);
 
-      uint64_t idx = d.executedSeqNum - ld.lastStableSeqNum;
+      uint64_t idx = d.executedSeqNum - ld.lastStableSeqNum - 1;
```

Comment:
Why was this change made?

if `d.executedSeqNum == 0 && ld.lastStableSeqNum == 0` then `idx` will underflow become UINT64_MAX and the Assert below will fire.

Reply:
Nevermind the last part of this. I see from the surrounding code (not shown in the PR) that this is safe as `d.executedSeqNum` must be greater than `ld.lastStableSeqNum.`

I'm still curious as to why this change is needed though. I just want to understand. I don't doubt it is correct.

Reply:
It just what we saw during the debugging. @guyg8  confirmed.